ci/fih_test_docker/fi_tester_gdb.sh

```diff
@@ -0,0 +1,200 @@
+#!/bin/bash
+
+# Copyright (c) 2020 Arm Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function skip_instruction {
+
+local SKIP_ADDRESS=$1
+local SKIP_SIZE=$2
+
+# Parse the ASM instruction from the address using gdb
+INSTR=$($GDB $AXF_FILE --batch -ex "disassemble $SKIP_ADDRESS" | grep "^ *$SKIP_ADDRESS" | sed "s/.*:[ \t]*\(.*\)$/\1/g")
+# Parse the C line from the address using gdb
+LINE=$($GDB $AXF_FILE --batch -ex "info line *$SKIP_ADDRESS" | sed "s/Line \([0-9]*\).*\"\(.*\)\".*/\2:\1/g")
+
+# Sometimes an address is in the middle of a 4 byte instruction. In that case
+# don't run the test
+if test "$INSTR" == ""; then
+    return
+fi
+
+# Print out the meta-info about the test, in YAML
+echo "- skip_test:"
+echo "    addr: $SKIP_ADDRESS"
+echo "    asm:  \"$INSTR\""
+echo "    line: \"$LINE\""
+echo "    skip: $SKIP_SIZE"
+# echo -ne "$SKIP_ADDRESS | $INSTR...\t"
+
+# Write out a script with the instructions for the debugger to run
+echo "
```

Comment:
I guess changing this to a here document would make it more readable.

```
cat >commands.gdb <<EOF
target remote localhost: 1234
b *$SKIP_ADDRESS
continue&
eval "shell sleep 0.5"
interrupt
if \$pc == $SKIP_ADDRESS
    echo "Stopped at breakpoint"
else
    echo "Failed to stop at breakpoint"
end
echo "PC before increase:"
print \$pc
set \$pc += $SKIP_SIZE
echo "PC after increase:"
print \$pc
detach
eval "shell sleep 0.5"
EOF
```

Reply:
Ok